src/io/github/mboegers/dop/invoice/MwStRechner.java

```diff
@@ -18,7 +18,9 @@ private PlainOOP() {
          * ermöglicht einen einheitlichen Zugriff
          */
         public static double calculateMwSt(Kunde kunde, double wert) {
-            return calculateMwSt(kunde, wert);
+            if (kunde instanceof Privatkunde p) return calculateMwSt(p, wert);
+            else if (kunde instanceof Businesskunde b) return calculateMwSt(b, wert);
+            else throw new IllegalArgumentException("Unsupported type: " + kunde.getClass());
```

Comment:
The idea of the MwStRechner.PlainOOP was to demonstrate a base implementation which is free of everything _modern_, like the Type Pattern. 

To have a simple single step experience, I favor the pure instanceof Operator in this place. This way a Type Cast is required, but the demo if the Type Pattern may benefit also. The usage of the Type Pattern is demonstrated in the next step MwStRechner.InstanceOfPattern#calculateMwSt.

Suggestion for line 21-23: 
´´´java
if (kunde instanceof Privatkunde) return calculateMwSt((Privatkunde) p, wert);
else if (kunde instanceof Businesskunde) return calculateMwSt((Businesskunde)b, wert);
else throw new IllegalArgumentException("Unsupported type: " + kunde.getClass());
´´´